### PR TITLE
Update and add support to Keychron Q1v1

### PR DESCRIPTION
**Pull request by adophoxia**
Add support for ANSI Knob and ISO variants of the Q1v1 while also updating the Q1v1 no-knob ANSI variant.

### CODE REVIEW

File: keyboards/keychron/q1/rev_0100/rules.mk [hunk 1]
```diff
@@ -19,3 +19,4 @@ AUDIO_ENABLE = no           # Audio output
 DIP_SWITCH_ENABLE = yes
 RGB_MATRIX_ENABLE = yes
 RGB_MATRIX_DRIVER = IS31FL3733
+LTO_ENABLE = yes
```
Comment (xyzz) [inline keyboards/keychron/q1/rev_0100/rules.mk]: Any changes should go under the `vial` keymap

File: keyboards/keychron/q1/rev_0101/rules.mk [hunk 2]
```diff
@@ -19,4 +19,5 @@ AUDIO_ENABLE = no           # Audio output
 ENCODER_ENABLE = yes        # Enable Encoder
 DIP_SWITCH_ENABLE = yes
 RGB_MATRIX_ENABLE = yes
-RGB_MATRIX_DRIVER = IS31FL3733
\ No newline at end of file
+RGB_MATRIX_DRIVER = IS31FL3733
+LTO_ENABLE = yes
```
Comment (xyzz) [inline keyboards/keychron/q1/rev_0101/rules.mk]: as above

File: keyboards/keychron/q1/rev_0102/rules.mk [hunk 3]
```diff
@@ -18,4 +18,5 @@ RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
 AUDIO_ENABLE = no           # Audio output
 DIP_SWITCH_ENABLE = yes
 RGB_MATRIX_ENABLE = yes
-RGB_MATRIX_DRIVER = IS31FL3733
\ No newline at end of file
+RGB_MATRIX_DRIVER = IS31FL3733
+LTO_ENABLE = yes
```
Comment (xyzz) [inline keyboards/keychron/q1/rev_0102/rules.mk]: as above

**Review by xyzz — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q1/rev_0100/rules.mk (hunk 1 above)
Comment (adophoxia) [reply keyboards/keychron/q1/rev_0100/rules.mk]: Made the necessary changes for LTO_ENABLE rule relocation.

File: keyboards/keychron/q1/rev_0101/rules.mk (hunk 2 above)
Comment (adophoxia) [reply keyboards/keychron/q1/rev_0101/rules.mk]: As above

File: keyboards/keychron/q1/rev_0102/rules.mk (hunk 3 above)
Comment (adophoxia) [reply keyboards/keychron/q1/rev_0102/rules.mk]: As above

File: keyboards/keychron/q1/rev_0100/rules.mk [hunk 4]
```diff
@@ -18,4 +18,4 @@ RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
 AUDIO_ENABLE = no           # Audio output
 DIP_SWITCH_ENABLE = yes
 RGB_MATRIX_ENABLE = yes
-RGB_MATRIX_DRIVER = IS31FL3733
+RGB_MATRIX_DRIVER = IS31FL3733
```
Comment (xyzz) [inline keyboards/keychron/q1/rev_0100/rules.mk:21]: there are still changes in this file

**Review by xyzz — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q1/rev_0100/rules.mk (hunk 4 above)
Comment (adophoxia) [reply keyboards/keychron/q1/rev_0100/rules.mk:21]: The only changes made was removing the "LTO_ENABLE" rule and moving it to the the keymap "rules.mk". I did check again to make sure that nothing else was affected besides that one line, to which I found nothing else changed.

**Comment by adophoxia** [comment]
Wanting to see if the changes requested in `rules.mk` is still needed.

**Pull request closed without merging**